### PR TITLE
Remove Vulkan renderer

### PR DESCRIPTION
**Pull request by TheSpydog**
Now that we have SDL_GPU, we can remove our old Vulkan renderer. And since it was the only backend that used FNA3D_Memory and FNA3D_CommandBuffer, we can delete those along with it!

Still need to test VisualC-GDK and Xcode, but the other build systems are happy with this change.

### CODE REVIEW

**Comment by flibitijibibo** [comment]
Marked as a draft until #219 is done - this will be removed in 25.01 at the latest but if threading support gets done early we could probably do this a month early...

File: visualc-gdk/FNA3D.vcxproj [hunk 1]
```diff
@@ -56,11 +56,6 @@
     <ClInclude Include="..\src\FNA3D_Driver.h" />
     <ClInclude Include="..\src\FNA3D_Tracing.h" />
   </ItemGroup>
-  <ItemGroup>
```
Comment (flibitijibibo) [inline visualc-gdk/FNA3D.vcxproj:59]: Should these references be removed as part of the move to SDL3?

Comment (TheSpydog) [reply visualc-gdk/FNA3D.vcxproj:59]: Yeah, we need some way of linking with both SDL2/3 (for now) so I'm not sure what this should look like. Though I don't remember making this specific change, the project reference might have been removed automatically by my IDE.

**Review by flibitijibibo — APPROVED** [review]
We'll probably want to readd the one GDK thing but other than that we're good to go on this.

**Comment by thatcosmonaut** [comment]
Good night, sweet prince, and flights of angels sing thee to thy rest. 
🪦👼

**Comment by flibitijibibo** [comment]
lol rip

**Pull request merged**